cmd/allocator/main.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	allocationv1alpha1 "agones.dev/agones/pkg/apis/allocation/v1alpha1"
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"agones.dev/agones/pkg/util/runtime"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	logger = runtime.NewLoggerWithSource("main")
+)
+
+const (
+	certDir = "/home/allocator/client-ca/"
+	tlsDir  = "/home/allocator/tls/"
+	port    = "8443"
+)
+
+// A handler for the web server
+type handler func(w http.ResponseWriter, r *http.Request)
+
+func main() {
+	agonesClient, err := getAgonesClient()
+	if err != nil {
+		logger.WithError(err).Fatal("could not create agones client")
+	}
+
+	h := httpHandler{
```

Comment:
Do we think we'll do a gRPC endpoint as well? Just curious what your take was.

Reply:
Yes. In the next iteration, I will add the gRPC.